### PR TITLE
Use github issue template forms

### PR DESCRIPTION
**Pull request by rjernst**
Currently we use GitHub issue templates containing markdown for each
template. This was an improvement over the single issue template before
that, and there are several bits of information that we ask for, but
filling out the template is cumbersome because it is completely
markdown, and users often leave bits of markdown comments meant for them
to read in the resulting issue.

This commit converts the existing issue templates to use the new github
issue template forms. These are configured through yml, and present an
html form to the user that is customized for each issue type.

### CODE REVIEW

**Comment by rjernst** [comment]
For edification, [here](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) are the docs on the issue template forms.

File: .github/ISSUE_TEMPLATE/test-failure.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      description: The reproduce line from the build output
+    validations:
+      required: true
+  - type: dropdown
+    id: reproducible
+    attributes:
+      label: Does it reproduce?
+      options:
+        - "Yes"
+        - "No"
```
Comment (mark-vieira) [inline .github/ISSUE_TEMPLATE/test-failure.yml:28]: FWIW, for the Slack automation we also include a "Didn't try" option.

File: .github/ISSUE_TEMPLATE/test-failure.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    validations:
+      required: true
+  - type: input
+    id: branches
+    attributes:
+      label: Applicable branches
+      description: Branch names this failure has occurred on 
+    validations:
+      required: false
```
Comment (mark-vieira) [inline .github/ISSUE_TEMPLATE/test-failure.yml:37]: I'd personally make this required. It doesn't require any additional effort from the submitter, they certainly observed this on _some_ branch.

File: .github/ISSUE_TEMPLATE/test-failure.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  - type: markdown
+    attributes:
+      value: Please fill out the following information, and ensure you have attempted to reproduce locally
+  - type: input
+    id: build_scan
+    attributes:
+      label: Build scan
+      description: A link to a Gradle buildscan with the failure
+    validations:
+      required: true
```
Comment (mark-vieira) [inline .github/ISSUE_TEMPLATE/test-failure.yml:14]: Not all failures necessarily produce a scan. Maybe we should reword this to optionally be a link to a Jenkins job in those scenarios where things exploded prior to a scan being created.

**Review by mark-vieira — APPROVED** [review]
Couple minor comments but LGTM. This is definitely an improvement over the markdown templates.

File: .github/ISSUE_TEMPLATE/test-failure.yml (hunk 1 above)
Comment (rjernst) [reply .github/ISSUE_TEMPLATE/test-failure.yml:28]: Added

File: .github/ISSUE_TEMPLATE/test-failure.yml (hunk 3 above)
Comment (rjernst) [reply .github/ISSUE_TEMPLATE/test-failure.yml:14]: I reworded this to allow a jenkins job link, and relabeled it to "CI Link".
